### PR TITLE
Cal 21

### PR DESCRIPTION
**Pull request by sylgas**

### CODE REVIEW

File: src/main/webapp/resources/views/right-panel.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class="page-content">
+    <h1>{{expandEventForm ? "Event" : "Event list"}}
+        <span ng-hide="expandEventForm" class="pull-right btn btn-link" ng-click="expandForm()">+ Add new event</span>
+    </h1>
+
+    <div class="form-group"></div>
```
Comment (pkrp) [inline src/main/webapp/resources/views/right-panel.html]: what for?

File: src/main/webapp/resources/scripts/controllers/RootController.coffee [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    position =
+      lat: $scope.form.event.location.longitude
+      lng: $scope.form.event.location.latitude
+    if position.lat and position.lng
+      $rootScope.$emit(Event.POINTER_POSITION_CHANGED, position)
+
+  setCoordinates = (position) ->
+    $scope.form.event.location =
+      longitude: parseFloat(position.lat().toFixed(7))
+      latitude: parseFloat(position.lng().toFixed(7))
```
Comment (pkrp) [inline src/main/webapp/resources/scripts/controllers/RootController.coffee]: got error: position.lat is not a function when filling manually coordinates

File: src/main/webapp/resources/index.html [hunk 3]
```diff
@@ -1,56 +1,76 @@
...
-<div class="container">
-    <div ng-view></div>
+<div ng-controller="RootController">
+    <div id="breadcrumb" ng-if="title" class="breadcrumb" style="margin-bottom: 0">
+        <label>{{title}}</label>
+        <span class="pull-right"><a href="" ng-click="panel.show = !panel.show">{{panel.show? "Hide events panel":
+            "Show events panel"}}</a></span>
+    </div>
+    <div class="no-padding" ng-class="panel.show? 'col-md-8': 'col-md-12'" ng-view></div>
+    <div class="col-md-4 no-padding page-slide" ng-show="panel.show" full-height
```
Comment (pkrp) [inline src/main/webapp/resources/index.html]: I would rather change width of panel to col-md-3. On the smaller screen it still looks nice

File: src/main/webapp/resources/index.html [hunk 4]
```diff
@@ -1,56 +1,76 @@
...
 </head>
 
 <body>
 <div>
-    <nav class="navbar navbar-inverse" role="navigation" style="margin-bottom: 0">
-        <div class="navbar-header" ng-click="title = ''">
-            <a class="navbar-brand" href="#"><span class="glyphicon glyphicon glyphicon-calendar"></span></a>
+    <nav id="navigation" class="navbar navbar-inverse" role="navigation">
+        <div class="navbar-header">
+            <a class="navbar-brand" href="#/"><span class="glyphicon glyphicon glyphicon-calendar"></span></a>
```
Comment (pkrp) [inline src/main/webapp/resources/index.html]: glyphicon glyphicon-calendar

File: src/main/webapp/resources/scripts/controllers/MapController.coffee [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
-        zoom: 15,
-        mapTypeId: google.maps.MapTypeId.ROADMAP
-      }
+angular.module('calendar').controller 'MapController', ($rootScope, $scope, Event) ->
+  $rootScope.title = "Map"
+
+  $scope.markers = []
+
+  hidePointer = ->
+    $scope.pointer.setMap(null)
```
Comment (pkrp) [inline src/main/webapp/resources/scripts/controllers/MapController.coffee]: pointer here is undefined when saving/cancelling event with initialized map

(add or edit event with map in background)

File: src/main/webapp/resources/scripts/controllers/RootController.coffee [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  saveEvent = (event) ->
+    EventService.save(event).then (saved) ->
+      if event.id
+        event = saved
+      else
+        $scope.events.push saved
+      $scope.collapseForm()
+
+  openEditFrom = (event) ->
```
Comment (pkrp) [inline src/main/webapp/resources/scripts/controllers/RootController.coffee:26]: openEditForm

File: src/main/webapp/resources/scripts/services/EventService.coffee [hunk 7]
```diff
@@ -0,0 +1,22 @@
+angular.module('calendar').service 'EventService', (Restangular, $q) ->
+  Events = Restangular.service('event')
+
+  fromBacked = (event) ->
```
Comment (pkrp) [inline src/main/webapp/resources/scripts/services/EventService.coffee]: fromBackend

File: src/main/webapp/resources/scripts/controllers/RootController.coffee (hunk 6 above)
Comment (sylgas) [inline src/main/webapp/resources/scripts/controllers/RootController.coffee:26]: it is also createForm and it is not opened like modal but extended

**Pull request merged**